### PR TITLE
tests: does not display errors in gen_opt_test.vim in windows gVim

### PR DESCRIPTION
**Pull request by Milly**
This allows we to see in the CI log if the test implementation is incorrect.

### CODE REVIEW

**Comment by Milly** [comment]
[I tested the error log in CI](https://github.com/Milly/vim/actions/runs/11372497876/job/31636949780#step:11:282):

```log
	..\gvim -e -s -u NONE -U NONE --noplugin --not-a-term --gui-dialog-file guidialog --nofork -S gen_opt_test.vim ../optiondefs.h

Error: Invalid option name in test_values: 'notexistoption' in command line..script D:\a\vim\vim\src\testdir\gen_opt_test.vim, line 200
```

File: src/testdir/Makefile [hunk 1]
```diff
@@ -160,8 +160,12 @@ test_gui_init.res: test_gui_init.vim
 	$(RUN_VIMTEST) -u gui_preinit.vim -U gui_init.vim $(NO_PLUGINS) -S runtest.vim $<
 	@rm vimcmd
 
-opt_test.vim: ../optiondefs.h gen_opt_test.vim
-	$(VIMPROG) -e -s -u NONE $(NO_INITS) --nofork --gui-dialog-file guidialog -S gen_opt_test.vim ../optiondefs.h
+opt_test.vim: gen_opt_test.vim ../optiondefs.h
+	$(VIMPROG) -e -s -u NONE $(NO_INITS) --nofork --gui-dialog-file guidialog -S $^
```
Comment (k-takata) [inline src/testdir/Makefile]: Is `$^` available in BSD make or in other makes?

Comment (k-takata) [reply src/testdir/Makefile]: It fails on FreeBSD: https://github.com/vim/vim/pull/15887/checks?check_run_id=31638502107

Comment (Milly) [reply src/testdir/Makefile]: Oops

Comment (Milly) [reply src/testdir/Makefile]: Fixed.

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**